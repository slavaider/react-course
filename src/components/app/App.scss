.App{
  .pagination{
    justify-content:center;
    margin-top: 10px;
  }
}
