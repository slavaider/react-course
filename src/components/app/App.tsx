import React from 'react';
import { BrowserRouter, Route, Switch } from 'react-router-dom';
import './App.scss';
import { Provider } from 'react-redux';
import AboutPage from '../../pages/about-page/AboutPage';
import HomePage from '../../pages/home-page/HomePage';
import ErrorPage from '../../pages/error-page/ErrorPage';
import Header from '../header/Header';
import DetailsPage from '../../pages/details-page/DetailsPage';
import store from '../../store/store';

const App: React.FC = () => {
  return (
    <Provider store={store}>
      <BrowserRouter>
        <div className="App">
          <Header />
          <main className="container mt-1">
            <Switch>
              <Route path="/" exact component={HomePage} />
              <Route path="/about" exact component={AboutPage} />
              <Route path="/details/:title" component={DetailsPage} />
              <Route path="*" component={ErrorPage} />
            </Switch>
          </main>
        </div>
      </BrowserRouter>
    </Provider>
  );
};

export default App;
