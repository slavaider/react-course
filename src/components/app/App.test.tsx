import React from 'react';
import { render } from '@testing-library/react';
import App from './App';

test('renders correctly', async () => {
  const wrapper = render(<App />);
  expect(wrapper).toMatchSnapshot();
});
