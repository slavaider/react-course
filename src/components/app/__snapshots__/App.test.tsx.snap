// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <nav
          aria-label="Third navbar example"
          class="navbar navbar-expand-sm navbar-dark bg-dark"
        >
          <div
            class="container-fluid"
          >
            <button
              aria-controls="navbarsExample03"
              aria-expanded="false"
              aria-label="Toggle navigation"
              class="navbar-toggler"
              data-bs-target="#navbarsExample03"
              data-bs-toggle="collapse"
              type="button"
            >
              <span
                class="navbar-toggler-icon"
              />
            </button>
            <div
              class="collapse navbar-collapse"
              id="navbarsExample03"
            >
              <ul
                class="navbar-nav me-auto mb-2 mb-sm-0"
              >
                <li
                  class="nav-item"
                >
                  <a
                    aria-current="page"
                    class="nav-link active"
                    href="/"
                  >
                    Home
                  </a>
                </li>
                <li
                  class="nav-item"
                >
                  <a
                    class="nav-link"
                    href="/about"
                  >
                    About
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </nav>
        <main
          class="container mt-1"
        >
          <div
            aria-label="transition"
            class="wrapper my-node-enter"
          >
            <div
              class="page-wrapper"
            >
              <div
                class="SearchBar row"
              >
                <div
                  class="searchbar"
                >
                  <form
                    aria-label="search-bar-form"
                    class="form"
                  >
                    <label
                      class="form-label"
                    >
                      <p
                        class="text-white"
                      >
                        Sort:
                      </p>
                      <select
                        aria-label="sort"
                        class="form-select"
                        name="sort"
                      >
                        <option
                          selected=""
                          value="publishedAt"
                        >
                          Published
                        </option>
                        <option
                          value="popularity"
                        >
                          Popularity
                        </option>
                        <option
                          value="relevancy"
                        >
                          Relevancy
                        </option>
                      </select>
                    </label>
                    <div
                      class="search-wrapper"
                    >
                      <input
                        aria-label="search"
                        class="form-control"
                        name="search"
                        placeholder="Search..."
                        required=""
                        type="search"
                      />
                      <button
                        aria-label="search-submit"
                        class="search_icon"
                        type="submit"
                      >
                        <i
                          class="fas fa-search"
                        />
                      </button>
                    </div>
                  </form>
                </div>
              </div>
              <div
                class="CardWrapper row"
              />
            </div>
          </div>
        </main>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <nav
        aria-label="Third navbar example"
        class="navbar navbar-expand-sm navbar-dark bg-dark"
      >
        <div
          class="container-fluid"
        >
          <button
            aria-controls="navbarsExample03"
            aria-expanded="false"
            aria-label="Toggle navigation"
            class="navbar-toggler"
            data-bs-target="#navbarsExample03"
            data-bs-toggle="collapse"
            type="button"
          >
            <span
              class="navbar-toggler-icon"
            />
          </button>
          <div
            class="collapse navbar-collapse"
            id="navbarsExample03"
          >
            <ul
              class="navbar-nav me-auto mb-2 mb-sm-0"
            >
              <li
                class="nav-item"
              >
                <a
                  aria-current="page"
                  class="nav-link active"
                  href="/"
                >
                  Home
                </a>
              </li>
              <li
                class="nav-item"
              >
                <a
                  class="nav-link"
                  href="/about"
                >
                  About
                </a>
              </li>
            </ul>
          </div>
        </div>
      </nav>
      <main
        class="container mt-1"
      >
        <div
          aria-label="transition"
          class="wrapper my-node-enter"
        >
          <div
            class="page-wrapper"
          >
            <div
              class="SearchBar row"
            >
              <div
                class="searchbar"
              >
                <form
                  aria-label="search-bar-form"
                  class="form"
                >
                  <label
                    class="form-label"
                  >
                    <p
                      class="text-white"
                    >
                      Sort:
                    </p>
                    <select
                      aria-label="sort"
                      class="form-select"
                      name="sort"
                    >
                      <option
                        selected=""
                        value="publishedAt"
                      >
                        Published
                      </option>
                      <option
                        value="popularity"
                      >
                        Popularity
                      </option>
                      <option
                        value="relevancy"
                      >
                        Relevancy
                      </option>
                    </select>
                  </label>
                  <div
                    class="search-wrapper"
                  >
                    <input
                      aria-label="search"
                      class="form-control"
                      name="search"
                      placeholder="Search..."
                      required=""
                      type="search"
                    />
                    <button
                      aria-label="search-submit"
                      class="search_icon"
                      type="submit"
                    >
                      <i
                        class="fas fa-search"
                      />
                    </button>
                  </div>
                </form>
              </div>
            </div>
            <div
              class="CardWrapper row"
            />
          </div>
        </div>
      </main>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
